bundles/org.eclipse.wst.xml.xpath2.processor/src/org/eclipse/wst/xml/xpath2/processor/internal/types/XSDuration.java

```diff
@@ -68,36 +80,54 @@
 	 */
 	public XSDuration(int years, int months, int days, int hours, int minutes,
 			BigDecimal seconds, boolean negative) {
-		_year = years;
-		_month = months;
-		_days = days;
-		_hours = hours;
-		_minutes = minutes;
-		_seconds = seconds;
-		_negative = negative;
-
-		if (_month >= 12) {
-			_year += _month / 12;
-			_month = _month % 12;
+		assert years >= 0;
+		assert months >= 0;
+		assert days >= 0;
+		assert hours >= 0;
+		assert minutes >= 0;
+		assert seconds != null && seconds.compareTo(BigDecimal.ZERO) >= 0;
```

Comment:
Are the asserts in the XSDuration constructor going to bite us?  ...or is the assumtion that there is no way for XSDuration to be called without those conditions holding true